code/datums/api.dm

```diff
@@ -1046,3 +1046,156 @@ proc/api_update_command_database()
 			statuscode = 500
 			response = "Orders have been placed. Use force parameter to overwrite."
 	return 1
+
+//Gets a overview of all polls (title, id, type)
+/datum/topic_command/get_polls
+	name = "get_polls"
+	description = "Gets a overview of all polls."
+	params = list(
+		"current_only" = list("name"="current_only","desc"="Only get information about the current polls","type"="int","req"=0),
+		"admin_only" = list("name"="admin_only","desc"="Only get information about the admin_only polls","type"="int","req"=0)
+	)
+
+/datum/topic_command/get_polls/run_command(queryparams)
+	var/current_only = text2num(queryparams["current_only"])
+	var/admin_only = text2num(queryparams["admin_only"])
+	
+	if(!dbcon.IsConnected())
```

Comment:
Use `establish_db_connection(dbcon)`. It has a chance of reconnecting the database upon a drop out or something.